src/test/addrman_tests.cpp

```diff
@@ -252,6 +252,28 @@ BOOST_AUTO_TEST_CASE(addrman_select_by_network)
     }
 }
 
+BOOST_AUTO_TEST_CASE(addrman_select_special)
+{
+    // use a non-deterministic addrman to ensure a passing test isn't due to setup
+    auto addrman = std::make_unique<AddrMan>(EMPTY_NETGROUPMAN, /*deterministic=*/false, GetCheckRatio(m_node));
+
+    // add ipv4 address to the new table
+    CNetAddr source = ResolveIP("252.2.2.2");
+    CService addr1 = ResolveService("250.1.1.3", 8333);
+    BOOST_CHECK(addrman->Add({CAddress(addr1, NODE_NONE)}, source));
+
+    // add I2P address to the tried table
+    CAddress i2p_addr;
+    i2p_addr.SetSpecial("udhdrtrcetjm5sxzskjyr5ztpeszydbh4dpl3pl4utgqqw2v4jna.b32.i2p");
+    BOOST_CHECK(addrman->Add({i2p_addr}, source));
```

Comment:
This non-deterministic test could fail intermittently, if `i2p_addr` collides with  `addr1` (I tried it, and it happened to me after ~4000 runs).
It would be good to change the order, such that we first add an address to tried (via new), and only add another addr to new when the first has been moved over to tried.